apps/ntag/ntagCmdApp/ntag_main.cpp

```diff
@@ -0,0 +1,428 @@
+                fprintf(stderr, "%s\n", blkHex);
+	        strcpy(ntagp, blkHex);
+	        ntagp += strlen(blkHex);
+	    }
+    	    else
+                fprintf(stderr, "Failed in call to le_hex_BinaryToString of Eeprom hex\n");
+    	}
+    }
+
+    dhubIO_PushString(TAG_MEM_EEPROM, DHUBIO_NOW, EepromHex);
+/*
+    // Let's read SRAM memory - for the NT3H2111_2211 blocks 0xF8 - 0xFB are sram
+    len = (0xFB - 0xF8 + 0x1) * NTAG_BLOCK_SIZE ;
+    if(!ntag.readSram(0, ntagp, len))
+    {
+        fprintf(stderr, "Failed in reading SRAM memory\n");
+	exit(2);
+    }
+
+    if(le_hex_BinaryToString(ntagp, len, EepromHex, sizeof(EepromHex)) != -1)
+        fprintf(stderr, "SRAM hex of the tag is: %s\n", EepromHex);
+    else
+        fprintf(stderr, "Failed in call to le_hex_BinaryToString\n");
+
+    ntagp += len;
+*/
+    dhubIO_PushString(TAG_MEM_SRAM, DHUBIO_NOW, "NOT IMPLEMENTED");
+
+    // Register reading is a byte based interface
+    for(i = 0 ; i <= 6 ; i++) {
+        if(!ntag.readRegister((Ntag::REGISTER_NR) i, reg)) {
+            fprintf(stderr, "Failed in reading Register: %d\n", i);
+	    exit(3);
+        }
+        else {
+	    //fprintf(stderr, "Session Register %d\t value: %X\n", i, reg);
+	    blk[i] = reg;
+        }
+    }
+
+    if(le_hex_BinaryToString(blk, 6, blkHex, sizeof(blkHex)) != -1)
+        fprintf(stderr, "Session Regs of the tag is: %s\n", blkHex);
+    else
+        fprintf(stderr, "Failed in call to le_hex_BinaryToString of Session Regs \n");
+
+    dhubIO_PushString(TAG_MEM_SESSION_REG, DHUBIO_NOW, blkHex);
+}
+
+//--------------------------------------------------------------------------------------------------
+COMPONENT_INIT
+{
+    // calling ntag without arguments will print the Usage
+    if (le_arg_NumArgs() < 1)
+    {
+        Usage();
+        exit(EXIT_SUCCESS);
+    }
+    else
+    {
+        const char *commandPtr = le_arg_GetArg(0);
+
+        if ((0 == strcmp(commandPtr, "help")) ||
+            (0 == strcmp(commandPtr, "--help")) ||
+            (0 == strcmp(commandPtr, "-h")))
+        {
+            Usage();
+            exit(EXIT_SUCCESS);
+        }
+        else
+        {
+            if (strcmp(commandPtr, "readndef") == 0)
+            {
+                ReadTagExtended();
+            }
+            else if (strcmp(commandPtr, "clean") == 0)
+            {
+                CleanTag();
+            }
+            else if (strcmp(commandPtr, "writeall") == 0)
+            {
+                WriteTagMultipleRecords();
+            }
+            else if (strcmp(commandPtr, "write") == 0)
+            {
+                WriteTag();
+            }
+            else if (strcmp(commandPtr, "read") == 0)
+            {
+                ReadTag();
+            }
+            else if (strcmp(commandPtr, "erase") == 0)
+            {
+                EraseTag();
+            }
+            else if (strcmp(commandPtr, "pushhubmemory") == 0)
```

Comment:
my preference would be "push_hub_memory", "push-hub-memory", "PushHubMemory" or "pushHubMemory" over "pushhubmemory" due to readability.  The same comment also applies to any other multi-word commands.